[Flaky Specs] Fix tag_rules and select2 issues

Matt-Yorkley opened the pull request:
#### What? Why?

Closes #6165 

Flaky specs: for some reason the select2 dropdown does not always close properly. The dropdown is still open when capybara moves on to the next step in the spec, which means it obscures other elements on the page, which can't be clicked. :boom:

The `tag_rules` feature spec currently has flaky failures on _multiple_ different lines, with two different underlying causes. I'm pretty sure this PR now fixes both of them, and should reduce general select2-related flakyness in all specs. 

#### What should we test?
<!-- List which features should be tested and how. -->

Green specs

#### Release notes
<!-- Write a one liner description of the change to be included in the release notes.
Every PR is worth mentioning, because you did it for a reason. -->

Fixed a flaky spec in the test suite

Changelog Category: Technical changes

Matt-Yorkley commented:
![BREAKING-BUILD](https://user-images.githubusercontent.com/9029026/96897685-26470700-148f-11eb-8ac9-7301453dc9d4.jpg)

sigmundpetersen commented:
5 🍏 in a row. Flakyness looking good 👌

sigmundpetersen commented:
https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/branches/master/builds/1824 😱

luisramos0 commented:
no need to panic :-) this PR is a good improvement anyway, we may need to further improve it.

Matt-Yorkley commented:
Yeah this spec was failing on ~8 different lines for two different reasons. I think I've fixed the select2-related failures (which was the worst) and the other part (tag lists) is hopefully improved somewhat, but it's also refactored in a way that should make it much easier to adjust if we need to take another crack at it.
